### PR TITLE
feat: track rosetta support before starting machine

### LINKED ISSUE #7798: Add rosetta precheck for macOS
### Is your enhancement related to a problem? Please describe

With 5.1 we need to add a precheck for rosetta as it seems for some reason it's not always enabled and ppl get 'vfkit exit 1 code'

### Describe the solution you'd like

add precheck

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
check if rosetta is working before starting the podman machine

### Screenshot / video of UI

image

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/7798

### How to test this PR?

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: extensions/podman/src/extension.ts [hunk 1]
```diff
@@ -749,8 +756,43 @@ export async function registerProviderFor(
   storedExtensionContext?.subscriptions.push(disposable);
 }
 
+export async function checkRosettaMacArm(podmanConfiguration: PodmanConfiguration): Promise<void> {
+  // check that rosetta is there for macOS / arm as the machine may fail to start
+  if (isMac() && os.arch() === 'arm64') {
+    const isEnabled = await podmanConfiguration.isRosettaEnabled();
+    if (isEnabled) {
+      console.log('checking if rosetta is enabled');
```
Comment (feloy) [inline extensions/podman/src/extension.ts]: to be removed?

**Review by feloy — APPROVED** [review]
I'm not able to remove rosetta to test this path.
LGTM for path when rosetta is installed and at code level

File: extensions/podman/src/extension.ts (hunk 1 above)
Comment (benoitf) [reply extensions/podman/src/extension.ts]: yes

**Comment by benoitf** [comment]
to try the removal of Rosetta I used https://gist.github.com/p-mng/af31593eb2a65f39baf9bcca9e4d2b8e

**Pull request merged**